### ISSUE TITLE
[bug] file association issues

### ISSUE THREAD

**Issue by thewh1teagle** [tauri-apps/tauri#9803]
### Describe the bug

I use file association in v2 app, and it works great in macOS. However in Linux it doesn't work at all and the app isn't suggested by right clicking some file.
In Windows it's partially works, with nsis: it doesn't list the app on right click but I also need to click "Choose another app" under the suggested apps context menu.
With msi it doesn't list it at all.

### Reproduction

Add file association to tauri app, compile and install in Linux/Windows and then right click some file with the desired extension 

### Expected behavior

It should list in suggested app just by right click some file and choosing open with

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 124.0.2478.97
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.77.1 (7cf61ebde 2024-03-27)
    ✔ cargo: 1.77.1 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.12.0
    - npm: 10.5.0
    - bun: 1.1.7

[-] Packages
    - tauri [RUST]: 2.0.0-beta.19
    - tauri-build [RUST]: 2.0.0-beta.15
    - wry [RUST]: 0.39.5
    - tao [RUST]: 0.28.0
    - @tauri-apps/api [NPM]: 2.0.0-beta.11
    - @tauri-apps/cli [NPM]: 2.0.0-beta.17

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

### Additional context

Repro open source app [vibe](https://github.com/thewh1teagle/vibe)

**Comment by martpie** [comment]
Also, the [file association example](https://github.com/tauri-apps/tauri/blob/dev/examples/file-associations) is pretty hard to follow and implement compared to other capabilities allowed by plugins. I wonder if there is a friendlier way of doing cross-platform file association?

**Comment by metkm** [comment]
Still, this is an issue. I can't make it work for some reason using msi on Windows. It doesn't get listed at all. It looks like using NSIS works but not MSI.

**Comment by metkm** [comment]
So, the workaround for this for now is to do it manually yourself and extend the .wxs file that was automatically generated. I'm going to put an example here.

tauri.conf.js

```json
    "windows": {
      "wix": {
        "fragmentPaths": ["./windows/fragments/registry.wxs"],
        "componentRefs": [
          "ContextDefaultIcon.mp4",
          "ContextOpenCommand.mp4",
          "ContextOpenProgids.mp4"
        ]
      }
    }
```

registry.wsx
```xml
<?xml version="1.0" encoding="utf-8"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
  <Fragment>
    <!-- these registry entries should be installed
     to the target user's machine -->
    <DirectoryRef Id="TARGETDIR">
      <!-- groups together the registry entries to be installed -->
      <!-- Note the unique `Id` we provide here -->

      <!-- MP4 -->
      <Component Id="ContextDefaultIcon.mp4" Guid="*">
        <RegistryKey
          Root="HKCU"
          Key="Software\Classes\myapp.mp4\DefaultIcon"
          Action="createAndRemoveOnUninstall">
          <RegistryValue
            Type="string"
            Value="[INSTALLDIR]myapp.exe,1"
            KeyPath="yes" />
        </RegistryKey>
      </Component>

      <Component Id="ContextOpenCommand.mp4" Guid="*">
        <RegistryKey
          Root="HKCU"
          Key="Software\Classes\myapp.mp4\shell\open\command"
          Action="createAndRemoveOnUninstall">
          <RegistryValue
            Type="string"
            Value="&quot;[INSTALLDIR]myapp.exe&quot; &quot;%1&quot;"
            KeyPath="yes" />
        </RegistryKey>
      </Component>

      <Component Id="ContextOpenProgids.mp4" Guid="*">
        <RegistryKey
          Root="HKCU"
          Key="Software\Classes\.mp4\OpenWithProgids"
          Action="createAndRemoveOnUninstall">
          <RegistryValue
            Type="string"
            Name="myapp.mp4"
            Value=""
            KeyPath="yes" />
        </RegistryKey>
      </Component>
    </DirectoryRef>
  </Fragment>
</Wix>
```

<img width="619" height="379" alt="Image" src="https://github.com/user-attachments/assets/0eb34c83-e51d-4ed8-a030-674cdda6c3b9" />
